src/exerciseCalculator.ts

```diff
@@ -0,0 +1,37 @@
+export interface ExerciseResult {
+    periodLength: number;
+    trainingDays: number;
+    success: boolean;
+    rating: number;
+    ratingDescription: string;
+    target: number;
+    average: number;
+}
+
+export const calculateExercises = (daily_exercises: number[], target: number): ExerciseResult => {
```

Comment:
Similar thing here, I would separate out the rating logic outside of this function